nomt/src/beatree/ops/update/leaf_stage.rs

```diff
@@ -35,50 +37,95 @@ pub struct ChangedLeafEntry {
     pub inserted: Option<LeafNode>,
 
     // the separator of the next node.
-    pub next_separator: Option<Key>,
+    next_separator: Option<Key>,
+}
+
+pub enum LeavesTrackerEntry {
+    ChangedLeaf(ChangedLeafEntry),
+    UnchangedRange(SeparatorRange),
```

Comment:
I'm ~ok with having this explicit, but IMO it is already implicit from the information in `changed`

Reply:
I think that removing it would make things less obvious but with cleaner code. I will add more comments to compensate for the absence of something explicit (in a follow-up)